frontends/web/src/components/sidebar/sidebar.tsx

```diff
@@ -92,175 +87,170 @@ const GetAccountLink = ({ coinCode, code, name, handleSidebarItemClick }: TGetAc
   );
 };
 
+const eject = (e: React.SyntheticEvent): void => {
+  apiPost('test/deregister');
+  e.preventDefault();
+};
 
-class Sidebar extends Component<Props> {
-  private swipe!: SwipeAttributes;
+const Sidebar = ({
+  deviceIDs,
+  accounts,
+  activeSidebar,
+  sidebarStatus, // from share HOC
+}: Props) => {
+  const { t } = useTranslation();
 
-  public componentDidMount() {
-    this.registerTouchEvents();
-  }
+  useEffect(() => {
+    const swipe = {
+      active: false,
+      x: 0,
+      y: 0,
+    };
 
-  public componentWillUnmount() {
-    this.removeTouchEvents();
-  }
+    const handleTouchStart = (event: TouchEvent) => {
+      const touch = event.touches[0];
+      swipe.x = touch.clientX;
+      swipe.y = touch.clientY;
+    };
 
-  private registerTouchEvents = () => {
-    document.addEventListener('touchstart', this.handleTouchStart);
-    document.addEventListener('touchmove', this.handleTouchMove);
-    document.addEventListener('touchend', this.handleTouchEnd);
-  };
+    const handleTouchMove = (event: TouchEvent) => {
+      if (
+        sidebarStatus !== 'forceHidden'
+        && event.changedTouches
+        && event.changedTouches.length
+      ) {
+        swipe.active = true;
+      }
+    };
 
-  private removeTouchEvents = () => {
-    document.removeEventListener('touchstart', this.handleTouchStart);
-    document.removeEventListener('touchmove', this.handleTouchMove);
-    document.removeEventListener('touchend', this.handleTouchEnd);
-  };
+    const handleTouchEnd = (event: TouchEvent) => {
+      if (sidebarStatus !== 'forceHidden') {
+        const touch = event.changedTouches[0];
+        const travelX = Math.abs(touch.clientX - swipe.x);
+        const travelY = Math.abs(touch.clientY - swipe.y);
+        const validSwipe = window.innerWidth <= 901 && swipe.active && travelY < 100 && travelX > 70;
+        if (
+          (!panelStore.state.activeSidebar && validSwipe && swipe.x < 60)
+          || (panelStore.state.activeSidebar && validSwipe && swipe.x > 230)
+        ) {
+          toggleSidebar();
+        }
+        swipe.x = 0;
+        swipe.y = 0;
+        swipe.active = false;
+      }
+    };
 
-  private handleTouchStart = (e: TouchEvent) => {
-    const touch = e.touches[0];
-    this.swipe = {
-      x: touch.clientX,
-      y: touch.clientY,
+    document.addEventListener('touchstart', handleTouchStart);
+    document.addEventListener('touchmove', handleTouchMove);
+    document.addEventListener('touchend', handleTouchEnd);
+    return () => {
+      document.removeEventListener('touchstart', handleTouchStart);
+      document.removeEventListener('touchmove', handleTouchMove);
+      document.removeEventListener('touchend', handleTouchEnd);
     };
-  };
+  }, [sidebarStatus]);
```

Comment:
this whole effect could probably become a hook, but keeping it here for now to not have too many changes.